static/app/components/organizations/environmentPageFilter/index.tsx

```diff
@@ -142,13 +151,13 @@ export function EnvironmentPageFilter({
         0
       );
 
-    // Calculate an appropriate width for the menu. It should be between 16 and 24em.
-    // Within that range, the width is a function of the length of the longest slug. The
-    // environment slugs take up to (longestSlugLength * 0.6)em of horizontal space (each
-    // character occupies roughly 0.6em). We also need to add 6em to account for the
-    // checkbox and menu paddings.
-    return `${Math.max(16, Math.min(24, longestSlugLength * 0.6 + 6))}em`;
-  }, [options]);
+    // Calculate an appropriate width for the menu. It should be between 16 (22 if
+    // there's a desynced message) and 24em. Within that range, the width is a function
+    // of the length of the longest slug. The environment slugs take up to
+    // (longestSlugLength * 0.6)em of horizontal space (each character occupies roughly
+    // 0.6em). We also need to add 6em to account for the checkbox and menu paddings.
+    return `${Math.max(desynced ? 22 : 16, Math.min(24, longestSlugLength * 0.6 + 6))}em`;
+  }, [options, desynced]);
```

Comment:
If the desync message is present, then the menu should be at least `22em` wide to accommodate it.